services/QuillLMS/app/services/google_integration/classrooms/updater.rb

```diff
@@ -0,0 +1,46 @@
+module GoogleIntegration
+  module Classrooms
+    class Updater
+      attr_reader :data, :google_classroom_id, :teacher_id
+
+      def initialize(data)
+        @data = data
+        @google_classroom_id = data[:google_classroom_id]
+        @teacher_id = data[:teacher_id]
+      end
+
+      def run
+        update
+        classroom
+      end
+
+      private def classroom
+        @classroom ||= ::Classroom.unscoped.find_by!(google_classroom_id: google_classroom_id, teacher_id: teacher_id)
+      end
+
+      private def custom_name?
+        classroom.name != classroom.synced_name
+      end
+
+      private def grade
+        data.key?(:grade) ? data[:grade] : classroom.grade
```

Comment:
Consider using fetch here (you can add a default value):
```suggestion
        data.fetch(:grade, classroom.grade)
```

Reply:
I was unaware of the default parameter.  That's much cleaner.